src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -768,24 +765,17 @@ AdaptiveExecutor(CitusScanState *scanState)
 		RunDistributedExecution(execution);
 	}
 
-	if (job->jobQuery->commandType != CMD_SELECT)
+	/* execute tasks local to the node (if any) */
+	if (list_length(execution->localTaskList) > 0)
 	{
-		if (list_length(execution->localTaskList) == 0)
-		{
-			Assert(executorState->es_processed == 0);
+		/* now execute the local tasks */
+		RunLocalExecution(scanState, execution);
+	}
 
-			executorState->es_processed = execution->rowsProcessed;
-		}
-		else if (distributedPlan->targetRelationId != InvalidOid &&
-				 !IsCitusTableType(distributedPlan->targetRelationId, REFERENCE_TABLE))
-		{
-			/*
-			 * For reference tables we already add rowsProcessed on the local execution,
-			 * this is required to ensure that mixed local/remote executions reports
-			 * the accurate number of rowsProcessed to the user.
-			 */
-			executorState->es_processed += execution->rowsProcessed;
-		}
+	CmdType commandType = job->jobQuery->commandType;
+	if (commandType != CMD_SELECT)
```

Comment:
This has been forever like this such that we don't set `executorState->es_processed ` for select queries. I couldn't find the reason yet.

Also, would that work fine with modifying ctes with SELECT query?

Reply:
iirc, it gives an unexpected extra output in psql if you set it. (like `SELECT 3`)